### ISSUE TITLE
Issue with Icons Related to Milestones

### ISSUE THREAD

**Issue by xlxCLUxlx** [Bungie-net/api#185]
Hi
I have run into an issue with icon data in the manifest (I did see the note about the pgcr issue but that seems to have been resolved a while ago for me).  I am currently using version 59912.17.09.13.1822-1.  I use the GetPublicMilestones endpoint to get the public milestones and then traverse the JSON and query the world_sql_content database to get relevant information that I care about in conjunction with live data from various endpoints.  I noticed that the icon property in the displayProperties for the manifest has an invalid GUID for the icon PNG.  This seems to only be related to the DestinyActivityDefinition, DestinyMilestoneDefinition and DestinyInventoryItemDefinition (when querying based on questHash for DestinyInventoryItemDefinition) tables.  I can combat this checking to see if the icon URL is valid or not and if not make a request to Destiny2/Armory/Search/DestinyMilestoneDefinition/nightfall/ for example and then compare the JSON result against the current weeks nighfall hash to get the PNG as the endpoint seems to return the correct GUID for the PNG.  This seems like unnecessary extra work though.  I may be missing something completely.  Any information would be appreciated.

Thanks for all the demanding work.  The OpenAPI spec so far has been very awesome to implement and I am loving the new milestone implementation so far over the previous advisors in Destiny 1.

**Comment by vthornheart-bng** [comment]
Thanks, and sorry for the long delay!  I still haven't had a chance to catch up on Github issues since my unexpected and extended absence - but I saw an email come in that linked to this issue, and took a look real quick to verify that indeed there seems to be something going wrong.  Flagging it now while I'm thinking about it so I know to loop back to it!

**Comment by xlxCLUxlx** [comment]
No worries...  I am able to work around it right now.  I have provided a workaround in issue [222](https://github.com/Bungie-net/api/issues/222) that is better than the initial workaround I was doing in this issue.

**Comment by vthornheart-bng** [comment]
How's this going for you these days?  Has this been resolved for you?

**Comment by xlxCLUxlx** [comment]
@vthornheart-bng ah, thanks for following up!  Honestly, I have not paid too much attention to this since I had a workaround in place in case I encountered a 404 error when trying to get the icon.  I did do a quick query against the `DestinyDefinitionMilestones` table though and it looks like all of the icons are accessible now without a 404 error.

```SQL
SELECT DISTINCT json_extract(DestinyMilestoneDefinition.json,'$.displayProperties.icon')
FROM DestinyMilestoneDefinition, json_tree(DestinyMilestoneDefinition.json, '$.displayProperties')
WHERE json_tree.key='icon';
```

I think this one is probably safe to close.  Thank you!
